src/openswathalgo/include/OpenMS/OPENSWATHALGO/ALGO/StatsHelpers.h

```diff
@@ -45,10 +46,8 @@ namespace OpenSwath
   template <typename Texp, typename Ttheo>
   double dotProd(Texp intExpBeg, Texp intExpEnd, Ttheo intTheo)
   {
-    std::vector<double> res(std::distance(intExpBeg, intExpEnd));
-    std::transform(intExpBeg, intExpEnd, intTheo, res.begin(), std::multiplies<double>());
-    double sum = std::accumulate(res.begin(), res.end(), 0.);
-    return sum;
+      return std::transform_reduce(std::execution::unseq, intExpBeg, intExpEnd,
```

Comment:
I deeply hate mac

Reply:
Yeah. ridiculous at this point..
Doesn't matter, we can just use Eigen. Should be equally, if not better vectorized